implementations/rust/ockam/ockam_command/src/authority/create.rs

```diff
@@ -0,0 +1,261 @@
+        args.push("--tenant-base-url".to_string());
+        args.push(tenant_base_url.clone());
+    }
+
+    if let Some(certificate) = &cmd.certificate {
+        args.push("--certificate".to_string());
+        args.push(certificate.clone());
+    }
+
+    if let Some(attributes) = &cmd.attributes {
+        args.push("--attributes".to_string());
+        args.push(attributes.join(","));
+    }
+    args.push(cmd.node_name.to_string());
+
+    run_ockam(opts, &cmd.node_name, args)
+}
+
+/// Start an authority node:
+///   - retrieve the node identity if the authority identity has been created before
+///   - persist the node state
+///   - start the node services
+async fn start_authority_node(
+    ctx: Context,
+    opts: (CommandGlobalOpts, CreateCommand),
+) -> crate::Result<()> {
+    let (options, cmd) = opts;
+    let command = cmd.clone();
+
+    // retrieve the authority identity if it has been created before
+    // otherwise create a new one
+    let public_identity = match options.state.identities.default().ok() {
+        Some(state) => state.config.public_identity(),
+        None => {
+            let cmd = identity::CreateCommand::new("authority".into(), None);
+            cmd.create_identity(ctx.async_try_clone().await?, options.clone())
+                .await?
+        }
+    };
+
+    let okta_configuration = match (
+        command.tenant_base_url,
+        command.certificate,
+        command.attributes,
+    ) {
+        (Some(tenant_base_url), Some(certificate), Some(attributes)) => Some(OktaConfiguration {
+            address: DefaultAddress::OKTA_IDENTITY_PROVIDER.to_string(),
+            tenant_base_url,
+            certificate,
+            attributes,
+        }),
+        _ => None,
+    };
+
+    // persist the node state and mark it as an authority node
+    // That flag allows the node to be seen as UP when listing the nodes with the
+    // the `ockam node list` command, without having to send a TCP query to open a connection
+    // because this would fail if there is no intention to create a secure channel
+    let node_state = options.state.nodes.get(&command.node_name)?;
+    let setup_config = node_state.setup()?;
+    node_state.set_setup(
+        &setup_config
+            .set_verbose(options.global_args.verbose)
+            .set_authority_node()
+            .add_transport(CreateTransportJson::new(
+                TransportType::Tcp,
+                TransportMode::Listen,
+                command.tcp_listener_address.as_str(),
+            )?),
+    )?;
+
+    let configuration = authority_node::Configuration {
+        identity: public_identity,
+        storage_path: options.state.identities.authenticated_storage_path()?,
+        vault_path: options.state.vaults.default()?.vault_file_path()?,
+        project_identifier: command.project_identifier,
+        tcp_listener_address: command.tcp_listener_address.clone(),
+        secure_channel_listener_name: None,
```

Comment:
This is just for the name of the secure channel listener service. I haven't left the possibility to customize it on the command line, so this is why there is `None` here which means the authority node will use `DefaultAddress::SECURE_CHANNEL_LISTENER` when starting the listener. If we decide to make this configurable from the command line, this will become a `Some(_)`